### ISSUE TITLE
TestApp.test_activate fails

### ISSUE THREAD

**Issue by Jeroendevr** [xlwings/xlwings#2261]
#### OS (e.g. Windows 10 or macOS Sierra)
macOS Ventura

#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)
xlwings - Latest(0.30.7)
Python - 3.9
Office 365
#### Describe your issue (incl. Traceback!)

Invoking the tests using pytest
```python
pytest tests/test_app.py::TestApp
```

Results in 
```python
platform darwin -- Python 3.9.16, pytest-7.2.2, pluggy-1.0.0 -- /Users/hayer/.virtualenvs/xlwings/bin/python
rootdir: /Users/hayer/xlwings/tests, configfile: pytest.ini
collected 17 items                                                                                                                                                                                                                                                                                                                       
====
tests/test_app.py::TestApp::test_activate FAILED                                                                                                                                                                                                                                                                                   [  5%]
tests/test_app.py::TestApp::test_books PASSED                                                                                                                                                                                                                                                                                      [ 11%]
tests/test_app.py::TestApp::test_calculation PASSED                                                                                                                                                                                                                                                                                [ 17%]
tests/test_app.py::TestApp::test_calculation_calculate PASSED                                                                                                                                                                                                                                                                      [ 23%]
tests/test_app.py::TestApp::test_display_alerts PASSED                                                                                                                                                                                                                                                                             [ 29%]
tests/test_app.py::TestApp::test_enable_events PASSED                                                                                                                                                                                                                                                                              [ 35%]
tests/test_app.py::TestApp::test_interactive SKIPPED (app.interactive is not supported on macOS)                                                                                                                                                                                                                                   [ 41%]
tests/test_app.py::TestApp::test_kill PASSED                                                                                                                                                                                                                                                                                       [ 47%]
tests/test_app.py::TestApp::test_len PASSED                                                                                                                                                                                                                                                                                        [ 52%]
tests/test_app.py::TestApp::test_macro PASSED                                                                                                                                                                                                                                                                                      [ 58%]
tests/test_app.py::TestApp::test_pid PASSED                                                                                                                                                                                                                                                                                        [ 64%]
tests/test_app.py::TestApp::test_quit PASSED                                                                                                                                                                                                                                                                                       [ 70%]
tests/test_app.py::TestApp::test_screen_updating PASSED                                                                                                                                                                                                                                                                            [ 76%]
tests/test_app.py::TestApp::test_selection PASSED                                                                                                                                                                                                                                                                                  [ 82%]
tests/test_app.py::TestApp::test_version PASSED                                                                                                                                                                                                                                                                                    [ 88%]
tests/test_app.py::TestApp::test_visible PASSED                                                                                                                                                                                                                                                                                    [ 94%]
tests/test_app.py::TestApp::test_wb_across_instances PASSED                                                                                                                                                                                                                                                                        [100%]
 TestApp.test_activate

self = <tests.test_app.TestApp testMethod=test_activate>, args = (), kwargs = {}

    def wrapped_method(self, *args, **kwargs):
        try:
>           return test_method(*args, **kwargs)

tests/common.py:28: 
====
tests/test_app.py:43: in test_activate
    self.assertEqual(self.app2, xw.apps.active)
E   AssertionError: <App [excel] 11880> != <App [excel] 11879>
=====
FAILED tests/test_app.py::TestApp::test_activate - AssertionError: <App [excel] 11880> != <App [excel] 11879>
 1 failed, 15 passed, 1 skipped in 29.69s 
░
```

**Comment by fzumstein** [comment]
Yes, the unit tests are in a bad state, see also: 

* https://github.com/xlwings/xlwings/blob/main/DEVELOPER_GUIDE.md#tests
* https://github.com/xlwings/xlwings/issues/1993

fixed in #2262

**Comment by Jeroendevr** [comment]
No problem, my current goal is to add typing. So I am running related tests to the `App` class. Because I am unfamiliar with the codebase, I don't feel comfortabele 😬 changing behaviour when the tests are not passing. So I'll probably add some more PR related to tests. 
We'll get there one commit at a time. 😀
